astropy/units/quantity.py

```diff
@@ -1303,15 +1303,11 @@ def __setitem__(self, i, value):
     # __contains__ is OK
 
     def __bool__(self):
-        """Quantities should always be treated as non-False; there is too much
-        potential for ambiguity otherwise.
         """
-        warnings.warn(
-            "The truth value of a Quantity is ambiguous. "
-            "In the future this will raise a ValueError.",
-            AstropyDeprecationWarning,
-        )
-        return True
+        Truthiness of quantities is ambigous, especially for logarithmic units and 
+        temperatures, thus we raise an error. Use explicit comparisons.
+        """
+        raise ValueError("The truth value of a Quantity is ambiguous.")
```

Comment:
I think `NotImplementedError` is perhaps more appropriate? 

Also, might as well have your very nice comment be part of the error message, e.g., "the truth value of a Quantity is ambiguous, especially for logarithmic units and temperatures."

Reply:
I started with `TypeError`, but the deprecation warning explicitly said `will raise a ValueError in the Future`

Reply:
OK, I should have seen that. Let's stick with `ValueError` then - that is also what `numpy` returns for arrays with more than one element.